fe/fe-core/src/main/java/com/starrocks/clone/DynamicPartitionScheduler.java

```diff
@@ -168,6 +167,16 @@ private ArrayList<AddPartitionClause> getAddPartitionClause(Database db, OlapTab
                         db.getOriginName(), olapTable.getName());
                 continue;
             }
+
+            for (Range<PartitionKey> partitionKeyRange : rangePartitionInfo.getIdToRange(false).values()) {
+                if (partitionKeyRange.contains(addPartitionKeyRange.lowerEndpoint()) &&
+                        addPartitionKeyRange.contains(partitionKeyRange.upperEndpoint()) &&
+                        !addPartitionKeyRange.upperEndpoint().equals(partitionKeyRange.upperEndpoint())) {
+                    addPartitionKeyRange =
+                            Range.closedOpen(partitionKeyRange.upperEndpoint(), addPartitionKeyRange.upperEndpoint());
+                }
+            }
+
             for (Range<PartitionKey> partitionKeyRange : rangePartitionInfo.getIdToRange(false).values()) {
                 // only support single column partition now
                 try {
```

Comment:
better to move 172~177 logical to 181 to reduce traverse rangePartitionInfo twice.
and add some comment for this logical.

Reply:
I have modified my code as you suggested.